test/Dialect/FIRRTL/lower-classes.mlir

```diff
@@ -471,3 +472,25 @@ firrtl.circuit "PathTargetReplaced" {
   firrtl.module private @WillBeReplaced(out %output: !firrtl.integer) {
   }
 }
+
+// CHECK-LABEL: firrtl.circuit "LocalPath"
+firrtl.circuit "LocalPath" {
+  // CHECK: hw.hierpath private [[MODULE_NLA:@.+]] [@Child]
+  // CHECK: hw.hierpath private [[WIRE_NLA:@.+]] [@Child::[[WIRE_SYM:@.+]]]
+
+  firrtl.module @Child() attributes {annotations = [{class = "circt.tracker", id = distinct[0]<>}]} {
+    // CHECK: firrtl.wire sym [[WIRE_SYM]]
+    %wire = firrtl.wire {annotations = [{class = "circt.tracker", id = distinct[1]<>}]} : !firrtl.uint<8>
+  }
+
+  firrtl.module @LocalPath() {
+    // CHECK: om.path_create instance %basepath [[MODULE_NLA]]
+    %0 = firrtl.path instance distinct[0]<>
+
+    // CHECK: om.path_create reference %basepath [[WIRE_NLA]]
+    %1 = firrtl.path reference distinct[1]<>
+
+    firrtl.instance child0 @Child()
+    firrtl.instance child1 @Child()
+  }
+}
```

Comment:
To elaborate on my comment... What I'm concerned about is if there is a way to rewrite this circuit such that `@Child` is duplicated into `@Child0` and `@Child1`. What are the trackers used then? After deduplication, it would seem like this would result in non-local annotation trackers which would then trip the error checking logic.

Reply:
I see. The main way I have noticed something like this happening is when PrefixModules clones modules. You can set up an example that would cause `@Child` to `@Child0` and `@Child1`, each with a local annotation with the same `distinct` ID. This would then error out in LowerClasses here: https://github.com/llvm/circt/blob/afd61f26eff984daf8807e3a57d79a4a93705f31/lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp#L585-L586

Which is equivalent to this existing check: https://github.com/llvm/circt/blob/afd61f26eff984daf8807e3a57d79a4a93705f31/lib/Dialect/FIRRTL/Transforms/EmitOMIR.cpp#L650-L654

Note that PrefixModules happens after Dedup, but it might be possible to hit a scenario where Dedup makes local annotations non-local, and that could potentially fail the single-instantiation check, which I think is what you're referring to.

I guess my thought is with this PR, we can support _some_ local annotations end-to-end, and in the cases where it doesn't work, we'd hit a well defined error case. So, this should support some cases, but it shouldn't allow any ambiguous cases through; those should still error in the expected ways.

Reply:
Does this help assuage your concerns? If the current state is fine, I'll go ahead and merge this PR.

Reply:
I have zero problems with this PR! Good to go!

I'm bringing up this problem with dedup because it's a general problem with any annotation representation. If this fix gets you further, please go for it. 👍 

Annotations are generally problematic and we're slowly tackling getting off them. You're aware of the problems.

Reply:
Yep makes sense, thanks. I think with this we'll at least be at parity with what we could do with the tracker annotations and EmitOMIR. Excited to design a more dataflow-y representation for paths.